### PR TITLE
Day 1 operation to replace OIDC server CA

### PR DESCRIPTION
**Pull request by jenting**
# Describe your changes

Supports day 1 operation to replace OIDC service certificates signed by custom/trusted certificate authority

# Related Issues / Projects
Please provide links to Bugzilla and other GitHub projects with your description if they are related to the changes.

Relates to: https://github.com/SUSE/avant-garde/issues/1767

### CODE REVIEW

File: adoc/admin-security-certificates.adoc [hunk 1]
```diff
@@ -344,63 +329,56 @@ openssl rsa -in /path/to/file.pem -out /path/to/file.key
...
+cp <CUSTOM_OIDC_CA_CERT_PATH> my-cluster/pki/oidc-ca.crt
+cp <CUSTOM_OIDC_CA_KEY_PATH> my-cluster/pki/oidc-ca.key
+chmod 644 my-cluster/pki/oidc-ca.crt
+chmod 600 my-cluster/pki/oidc-ca.key
+
+sed -i "s|oidc-ca-file: .*|oidc-ca-file: /etc/kubernetes/pki/oidc-ca.crt|" kubeadm-init.conf
 ----
++
+After this process, bootstrap the cluster with `skuba node bootstrap`.
+skuba uploads the local OIDC CA certificate to remote path specified in the local file `kubeadm-init.conf` key `oidc-ca-file`. At the time installing the add-on, skuba generates the OIDC server certificates and keys which are signed by the provided OIDC CA certificate and key pair and stored to its Secret resource.
```
Comment (innobead) [inline adoc/admin-security-certificates.adoc:354]:
```suggestion
skuba uploads the local OIDC CA certificate to the remote path specified in the local file `kubeadm-init.conf` key `oidc-ca-file`. When installing the add-on, skuba generates the OIDC server certificates and keys which are signed by the provided OIDC CA certificate and key pair and then stored to its Secret resource.
```

File: adoc/admin-security-certificates.adoc [hunk 2]
```diff
@@ -344,63 +329,56 @@ openssl rsa -in /path/to/file.pem -out /path/to/file.key
...
 
-* Replacing the `front-end proxy` CA certificate:
+** Without a Trusted CA Key:
++
+Use command `skuba cert generate-csr` to generates the OIDC server CSRs and keys in `<my-cluster>/pki` folder.
```
Comment (innobead) [inline adoc/admin-security-certificates.adoc:358]:
```suggestion
Use command `skuba cert generate-csr` to generate the OIDC server CSRs and keys in `<my-cluster>/pki` folder.
```

File: adoc/admin-security-certificates.adoc [hunk 3]
```diff
@@ -344,63 +329,56 @@ openssl rsa -in /path/to/file.pem -out /path/to/file.key
...
+After the CA signed the CSRs and issued the server certificates, puts the OIDC CA certificate and the OIDC server certificates in `<my-cluster>/pki` folder.
```
Comment (innobead) [inline adoc/admin-security-certificates.adoc:359]:
```suggestion
After the CA signs the CSRs and issued the server certificates, put the OIDC CA certificate and the OIDC server certificates in `<my-cluster>/pki` folder.
```

File: adoc/admin-security-certificates.adoc [hunk 4]
```diff
@@ -344,63 +329,56 @@ openssl rsa -in /path/to/file.pem -out /path/to/file.key
...
+cp <CUSTOM_OIDC_DEX_SERVER_CERT_PATH> my-cluster/pki/oidc-dex-server.crt
+cp <CUSTOM_OIDC_GANGWAY_SERVER_CERT_PATH> my-cluster/pki/oidc-gangway-server.crt
+chmod 644 my-cluster/pki/oidc-ca.crt
+chmod 644 my-cluster/pki/oidc-dex-server.crt
+chmod 644 my-cluster/pki/oidc-gangway-server.crt
 ----
-
++
 After this process, bootstrap the cluster with `skuba node bootstrap`.
+skuba uploads the local OIDC CA certificate to remote path specified in the local file `kubeadm-init.conf` key `oidc-ca-file`. At the time installing the add-on, skuba uploads the OIDC CA certificate and OIDC server certificate and key pair to its Secret resource.
```
Comment (innobead) [inline adoc/admin-security-certificates.adoc:373]:
```suggestion
skuba uploads the local OIDC CA certificate to the remote path specified in the local file `kubeadm-init.conf` key `oidc-ca-file`. At the time installing the add-on, skuba uploads the OIDC CA certificate and OIDC server certificate and key pair to its Secret resource.
```

File: adoc/admin-security-certificates.adoc [hunk 5]
```diff
@@ -344,63 +329,56 @@ openssl rsa -in /path/to/file.pem -out /path/to/file.key
...
-----
-mkdir -p my-cluster/pki/etcd
-cp <CUSTOM_ETCD_CA_CERT_PATH> my-cluster/pki/etcd/ca.crt
-cp <CUSTOM_ETCD_CA_KEY_PATH> my-cluster/pki/etcd/ca.key
-chmod 644 my-cluster/pki/etcd/ca.crt
-chmod 600 my-cluster/pki/etcd/ca.key
-----
+{productname} uses `oidc-dex` and `oidc-gangway` servers to do authentication and authorization.
+Administrators might choose to replace these server's certificates by issuing a trusted CA certificate after cluster deployment.
+This way, the user does not have to add specific certificates to their trusted keychain.
```
Comment (innobead) [inline adoc/admin-security-certificates.adoc:336]: Just revisited, a littlie confusing. 

`This way` means `automatic certificates genreated by skuba` or `replacing the generated ones with custom`. I assume you mean the first one?

Comment (jenting) [reply adoc/admin-security-certificates.adoc:336]: nice catch.  It'll be changed as
```
Administrators might choose to replace these server's certificates issued by the custom _or_ trusted CA certificate.
This way, the user does not have to add specific certificates to their trusted keychain.
```

**Pull request merged**